src/components/vm/vmRenameDialog.jsx

```diff
@@ -40,7 +40,12 @@ export const RenameDialog = ({ vmName, vmId, connectionName, toggleModal }) => {
             return;
 
         return domainRename({ connectionName, id: vmId, newName: newName })
-                .then(toggleModal, exc => {
+                .then(() => {
+                    toggleModal();
+                    // If we are on the VMs details page change the URL to reflect the new name after the rename operation succeeded
+                    if (cockpit.location.path.length > 0)
+                        cockpit.location.go(["vm"], Object.assign(cockpit.location.options, { name: newName, connection: connectionName }));
```

Comment:
This looks dangerous -- the [docs](https://cockpit-project.org/guide/latest/cockpit-location.html) say that cockpit.location can't be modified in-place, but that's what Object.assign does. I suggest to use the spread operator: `{ ...cockpit.location.options, name: newName, connection: connectionName }`, which will create a new object.